softlayer/vm/ubuntu.go

```diff
@@ -0,0 +1,324 @@
+package vm
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"html/template"
+	"net"
+	"net/http"
+	"time"
+)
+
+type Route struct {
+	Network string
+	Netmask string
+	Gateway string
+}
+
+type Interface struct {
+	Name           string
+	Auto           bool
+	AllowHotplug   bool
+	Address        string
+	Netmask        string
+	DefaultGateway bool
+	Gateway        string
+	Routes         []Route
+	DNS            []string
+}
+
+type Interfaces []Interface
+
+// first 2 and last reserved if secondary
+type Subnet struct {
+	NetworkIdentifier string `json:"networkIdentifier"`
+	Gateway           string `json:"gateway"`
+	BroadcastAddress  string `json:"broadcastAddress"`
+	Netmask           string `json:"netmask"`
+}
+
+func (s Subnet) Contains(address string) bool {
+	ipNet := net.IPNet{
+		IP:   net.ParseIP(s.NetworkIdentifier),
+		Mask: net.IPMask(net.ParseIP(s.Netmask)),
+	}
+
+	return ipNet.Contains(net.ParseIP(address))
+}
+
+type Subnets []Subnet
+
+func (s Subnets) Containing(address string) (Subnet, error) {
+	for _, subnet := range s {
+		if subnet.Contains(address) {
+			return subnet, nil
+		}
+	}
+
+	return Subnet{}, fmt.Errorf("subnet not found for %q", address)
+}
+
+type NetworkVLAN struct {
+	Name    string  `json:"name"`
+	Subnets Subnets `json:"subnets"`
+}
+
+type NetworkComponent struct {
+	Name             string      `json:"name"`
+	Port             int         `json:"port"`
+	PrimaryIPAddress string      `json:"primaryIpAddress"`
+	NetworkVLAN      NetworkVLAN `json:"networkVlan"`
+}
+
+type VirtualGuestNetworkComponents struct {
+	PrimaryBackendNetworkComponent NetworkComponent `json:"primaryBackendNetworkComponent,omitempty"`
+	PrimaryNetworkComponent        NetworkComponent `json:"primaryNetworkComponent,omitempty"`
+}
+
+//go:generate counterfeiter -o fakes/fake_softlayer_client.go . softLayerClient
+type softLayerClient interface {
+	DoRawHttpRequest(path string, requestType string, requestBody *bytes.Buffer) ([]byte, int, error)
+}
+
+//go:generate counterfeiter -o fakes/fake_sl_file_service.go --fake-name FakeSLFileService . softLayerFileService
+type softLayerFileService interface {
+	Upload(user string, password string, target string, destinationPath string, contents []byte) error
+}
+
+//go:generate counterfeiter -o fakes/fake_ssh_client.go --fake-name FakeSSHClient . sshClient
+type sshClient interface {
+	Output(cmd string) ([]byte, error)
+}
+
+type Ubuntu struct {
+	SoftLayerClient      softLayerClient
+	SSHClient            sshClient
+	SoftLayerFileService softLayerFileService
+}
+
+func SoftlayerPrivateRoutes(gateway string) []Route {
+	return []Route{
+		{Network: "10.0.0.0", Netmask: "255.0.0.0", Gateway: gateway},
+		{Network: "161.26.0.0", Netmask: "255.255.0.0", Gateway: gateway},
+	}
+}
+
+func (u *Ubuntu) ConfigureNetwork(networks Networks, vm VM) error {
```

Comment:
Already included in ubuntu_test.go
[ConfigureNetwork](https://github.com/cloudfoundry/bosh-softlayer-cpi/blob/develop_manual_networking/softlayer/vm/ubuntu_test.go#L59)